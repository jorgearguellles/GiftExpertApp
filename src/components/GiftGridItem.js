import PropTypes from 'prop-types';

const GiftGridItem = ({ title, urlImg }) => {

  return(
    <div className="card animate__animated animate__zoomIn">
      <img alt={title} src={urlImg} />
      <p>{title}</p>
    </div>
  )
};

export { GiftGridItem };

GiftGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  urlImg: PropTypes.string.isRequired,
}

GiftGridItem.defaultProps = {
  title: "Title here",
  urlImg: "https://www.google.com",
}

