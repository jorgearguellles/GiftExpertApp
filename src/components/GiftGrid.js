import { useFetchGifts } from "../hooks/useFetchGifts";
import { GiftGridItem } from "./GiftGridItem";
import PropTypes from 'prop-types';

const GiftGrid = ({ category }) => {

  const {data:images, loading} = useFetchGifts(category);

  return(
    <>
      <h4>{category}</h4>

      {loading && <p>Loading...</p>}
      
      <div className="card-grid">
        {
          images.map( img => ( <GiftGridItem key={img.id} {...img} />) )
        }
      </div> 
    </>
  )
};

export { GiftGrid };

GiftGrid.propTypes = {
  category: PropTypes.string.isRequired
}