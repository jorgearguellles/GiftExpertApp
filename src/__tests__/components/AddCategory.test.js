import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import {AddCategory} from "../../components/AddCategory";

describe("<AddCategory /> testing", ()=>{
  
  const setCategories = jest.fn(); // Jest functionality to make a function behavior. Now we can know when, how many was called
  let wrapper = shallow(<AddCategory setCategories={setCategories}/>);
  
  beforeEach(()=>{
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories}/>);
  });

  test("1. Should to render correctly", ()=>{
    expect(wrapper).toMatchSnapshot();
  });

  test("2. Should to change text input", ()=>{
    const input = wrapper.find('input');
    const value = "Hello World";

    // We need simulate event inside handleChange
    input.simulate('change', { target: { value } });
  });

  test("3. Shouldn't post info by submit", ()=>{
    wrapper.find('form').simulate('submit', { preventDefault(){} })
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("4. Should call setCategories and clean the text box", ()=>{

    //1.  Simulate input Change
    const value = "Hello World";
    wrapper.find('input').simulate('change', { target: { value } });  
    //2.  Simulated submit 
    wrapper.find('form').simulate('submit', { preventDefault(){} }) 
    // 3. SetCategories should been called
    expect(setCategories).toHaveBeenCalled(); 
    expect(setCategories).toHaveBeenCalledTimes(1); 
    // 4. Box text have to be a ''
    const textBoxValue = wrapper.find('input').prop('value');
    expect( textBoxValue ).toBe('');
  });
})