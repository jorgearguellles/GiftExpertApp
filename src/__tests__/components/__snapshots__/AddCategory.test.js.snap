// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory /> testing 1. Should to render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
